### ISSUE TITLE
Issue with rotation when rearviewcontroller visible

### ISSUE THREAD

**Issue by ebalassanian** [John-Lluch/SWRevealViewController#616]
Hello, 

I am having an issue with autorotation. I have configured the rearviewcontroller to take over the entire screen when it is visible. This works great expect when I rotate (which I need to support), the rearviewcontroller and the frontviewcontroller split the screen. I need to have the rearviewcontroller continue to occupy the screen bounds when rotated. Any suggestions?

**Comment by iDevelopper** [comment]
I did not found the solution but you can do it easily with https://github.com/iDevelopper/PBRevealViewController,
a librairie I wrote as SWRevealViewController seems not to be maintained anymore.

**Comment by ebalassanian** [comment]
Hi Patrick, 

Thanks for the response. I just tried plugging in PBRevealViewController. I had the same result. When I rotate the device when the left viewcontroller is present and taking over the full screen, the rotation causes the left view controller to take up half the screen. Ideally what I would like is to have the rotation result in the leftviewcontroller still taking over the full dimensions. Any suggestions? 

Edward

**Comment by iDevelopper** [comment]
Yes but I know what to do:

```
    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
        coordinator.animateAlongsideTransition({ (context:UIViewControllerTransitionCoordinatorContext) in
            self.revealViewController().leftViewRevealWidth = UIScreen.mainScreen().bounds.width

        }) { (context:UIViewControllerTransitionCoordinatorContext) in
        }
        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
    }
```

**Comment by ebalassanian** [comment]
Yes, I did try that right after emaiing you and it worked great. I calculated the ratio and applied it to the left view. Thanks!

**Comment by iDevelopper** [comment]
[Example3Swift.zip](https://github.com/John-Lluch/SWRevealViewController/files/443112/Example3Swift.zip)

You have to add a button on the left view controller if you want to close the menu without selecting an item.

**Comment by iDevelopper** [comment]
If you decide to continue with PB RevealViewController, I would like you to open issues in the PBRevealViewController's Github repository as this is not the right place here now! Thanks a lot!

**Comment by ebalassanian** [comment]
Of course.

**Comment by John-Lluch** [comment]
Hi iDeveloper, I am not currently able to give as much support on the class as I used to do (health issues), but please stop pushing your own controller for developer code issues that are anyway not related with the SWRevealViewController. This controller is very well supported on the web and used by thousands, so it's not right that you just tell people to replace it by your own. If you really think that yours is superior at least give some credit to me, as your code is blatantly based on my code.

**Comment by ebalassanian** [comment]
Hi John, for what it’s worth, I solved the problem and am continuing to use SWRevealViewController.

**Comment by iDevelopper** [comment]
Great!

**Comment by John-Lluch** [comment]
Hi ebalassanian. Having choice is good, so iDeveloper's controller is very welcome. My point was just that some of the developer issues are not this controller fault. Regards.
